forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-mock-store/forgerock-openbanking-uk-aspsp-rs-mock-store-server/src/main/java/com/forgerock/openbanking/aspsp/rs/store/api/openbanking/payment/v3_1_4/domesticpayments/DomesticPaymentsApiController.java

```diff
@@ -25,25 +25,154 @@
+                .updated(new Date())
+                .idempotencyKey(xIdempotencyKey)
+                .obVersion(VersionPathExtractor.getVersionFromPath(request))
+                .build();
+        frPaymentSubmission = new IdempotentRepositoryAdapter<>(domesticPaymentSubmissionRepository)
+                .idempotentSave(frPaymentSubmission);
+        return ResponseEntity.status(HttpStatus.CREATED).body(entityInstance(frPaymentSubmission, paymentConsent));
+    }
+
+    @Override
+    public ResponseEntity getDomesticPaymentsDomesticPaymentId(
+            String domesticPaymentId,
+            String authorization,
+            DateTime xFapiAuthDate,
+            String xFapiCustomerIpAddress,
+            String xFapiInteractionId,
+            String xCustomerUserAgent,
+            HttpServletRequest request,
+            Principal principal
+    ) throws OBErrorResponseException {
+        Optional<FRDomesticPaymentSubmission> isPaymentSubmission = domesticPaymentSubmissionRepository.findById(domesticPaymentId);
+        if (!isPaymentSubmission.isPresent()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Payment submission '" + domesticPaymentId + "' can't be found");
+        }
+        FRDomesticPaymentSubmission frPaymentSubmission = isPaymentSubmission.get();
 
-    public DomesticPaymentsApiController(DomesticConsentRepository domesticConsentRepository,
-                                         DomesticPaymentSubmissionRepository domesticPaymentSubmissionRepository,
-                                         ResourceLinkService resourceLinkService) {
-        super(domesticConsentRepository, domesticPaymentSubmissionRepository, resourceLinkService);
+        Optional<FRDomesticConsent> isPaymentSetup = domesticConsentRepository.findById(domesticPaymentId);
+        if (!isPaymentSetup.isPresent()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Payment setup behind payment submission '" + domesticPaymentId + "' can't be found");
+        }
+        FRDomesticConsent frPaymentSetup = isPaymentSetup.get();
+        return ResponseEntity.ok(entityInstance(frPaymentSubmission, frPaymentSetup));
     }
 
     @Override
+    public ResponseEntity<OBWritePaymentDetailsResponse1> getDomesticPaymentsDomesticPaymentIdPaymentDetails(
+            String domesticPaymentId,
+            String authorization,
+            DateTime xFapiAuthDate,
+            String xFapiCustomerIpAddress,
+            String xFapiInteractionId,
+            String xCustomerUserAgent,
+            HttpServletRequest request,
+            Principal principal
+    ) throws OBErrorResponseException {
+        // Optional endpoint - not implemented
+        return new ResponseEntity<OBWritePaymentDetailsResponse1>(HttpStatus.NOT_IMPLEMENTED);
+    }
+
+    private OBWriteDomesticResponse4 entityInstance(FRDomesticPaymentSubmission frPaymentSubmission, FRDomesticConsent frDomesticConsent) {
```

Comment:
I know I mentioned this before, but I'm really struggling with this method name :-)

Reply:
I will rename this method to `responseEntity` could be better name.

Reply:
I think that would be better, but I still have no problem with `packageResponse()`! ;-)

Also, won't this mean all controllers prior to v3.1.4 have `packageResponse()` and all of the ones afterwards have a different name? I don't see a need for the deviation, but I'm not going to lose sleep over it! :-)

Reply:
All versions use the same method name.

Reply:
Have I missed something? It looks like all versions prior to v3.1.4 have `packagePayment()` and 3.1.4 onwards have `responseEntity()`?!

Reply:
Fixed.